### PR TITLE
Tag YAML v0.2.0 [https://github.com/dcjones/YAML.jl]

### PR DESCRIPTION
**Pull request by sglyon**
Added support for reading YAML files with custom tags

### CODE REVIEW

File: YAML/versions/0.2.0/requires [hunk 1]
```diff
@@ -0,0 +1,4 @@
+julia 0.3
+Codecs
+Compat
```
Comment (tkelman) [inline YAML/versions/0.2.0/requires]: needs a minimum version

File: YAML/versions/0.2.0/requires [hunk 2]
```diff
@@ -0,0 +1,4 @@
+julia 0.3
```
Comment (tkelman) [inline YAML/versions/0.2.0/requires]: metadata is frozen for package versions supporting 0.3

Comment (tkelman) [inline YAML/versions/0.2.0/requires]: make sure travis runs on all supported julia versions - "release" is 0.5 and I plan on deprecating it, better to be explicit about version numbers

**Comment by sglyon** [comment]
Done, done, and done -- thanks

File: YAML/versions/0.2.0/requires [hunk 3]
```diff
@@ -0,0 +1,4 @@
+julia 0.4
+Codecs
+Compat 0.12
```
Comment (tkelman) [inline YAML/versions/0.2.0/requires]: I don't see anything newer than Compat.String, what are you using that was introduced in 0.12?

Comment (tkelman) [inline YAML/versions/0.2.0/requires]: oh probably tripped up by https://github.com/JuliaLang/Compat.jl/commit/7eeef71f1db0e70f3c19402975535f208ad4ebcc, that was a readme-only change which breaks the blame method, can click the new button to see the blame prior to that commit

**Comment by sglyon** [comment]
Ahh yep, the README only change tripped me up. Should be g2g now

**Comment by tkelman** [comment]
failing tests on julia 0.4 - https://travis-ci.org/dcjones/YAML.jl/jobs/202428710

**Comment by sglyon** [comment]
Good catch -- that didn't show up the other day. 

0.4 support has been removed.

**Comment by tkelman** [comment]
the travis failure
```
ERROR: LoadError: packages with unsatisfiable requirements found:
    Mads v0.3.2 – no valid versions exist for package FileIO
    Mads v0.3.4 – no valid versions exist for package FileIO
    Mads v0.3.5 – no valid versions exist for package FileIO
    Mads v0.3.6 – no valid versions exist for package FileIO
 in check_metadata(::Set{String}) at /home/travis/.julia/v0.5/PkgDev/src/entry.jl:301
 in check_metadata() at /home/travis/.julia/v0.5/PkgDev/src/entry.jl:288
 in include_from_node1(::String) at ./loading.jl:488
 in eval(::Module, ::Any) at ./boot.jl:234
 in process_options(::Base.JLOptions) at ./client.jl:239
 in _start() at ./client.jl:318
while loading /home/travis/build/JuliaLang/METADATA/.test/METADATA.jl, in expression starting on line 248
```
should be worked around by #8006

**Pull request merged**